ts/features/cieLogin/components/CieIdLoginWebView.tsx

```diff
@@ -0,0 +1,169 @@
+import { useIONavigation } from "../../../navigation/params/AppParamsList";
+import { getCieIDLoginUri, SpidLevel } from "../utils";
+import { useLollipopLoginSource } from "../../lollipop/hooks/useLollipopLoginSource";
+import LoadingSpinnerOverlay from "../../../components/LoadingSpinnerOverlay";
+import { useIODispatch, useIOSelector } from "../../../store/hooks";
+import { loginSuccess } from "../../../store/actions/authentication";
+import { SessionToken } from "../../../types/SessionToken";
+import ROUTES from "../../../navigation/routes";
+import { loggedInAuthSelector } from "../../../store/reducers/authentication";
+import { IdpSuccessfulAuthentication } from "../../../components/IdpSuccessfulAuthentication";
+
+export type WebViewLoginNavigationProps = {
+  spidLevel: SpidLevel;
+  isUat: boolean;
+};
+
+const iOSUserAgent =
+  "Mozilla/5.0 (iPhone; CPU iPhone OS 14_0_1 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Mobile/15E148 Safari/604.1";
+const defaultUserAgent = Platform.select({
+  ios: iOSUserAgent,
+  default: undefined
+});
+
+const originSchemasWhiteList = ["https://*", "iologin://*"];
+
+export type CieIdLoginProps = {
+  spidLevel: SpidLevel;
+  isUat: boolean;
+};
+
+const CieIdLoginWebView = ({ spidLevel, isUat }: CieIdLoginProps) => {
+  const navigation = useIONavigation();
+  const webView = useRef<WebView>(null);
+  const dispatch = useIODispatch();
+  const [authenticatedUrl, setAuthenticatedUrl] = useState<string | null>(null);
+  const loggedInAuth = useIOSelector(loggedInAuthSelector, _isEqual);
+  const loginUri = getCieIDLoginUri(spidLevel, isUat);
+
+  const navigateToCieIdAuthenticationError = useCallback(() => {
+    navigation.navigate(ROUTES.AUTHENTICATION, {
+      screen: ROUTES.AUTHENTICATION_CIE_ID_ERROR
+    });
+  }, [navigation]);
+
+  const { shouldBlockUrlNavigationWhileCheckingLollipop, webviewSource } =
+    useLollipopLoginSource(navigateToCieIdAuthenticationError, loginUri);
+
+  useEffect(() => {
+    // https://reactnative.dev/docs/linking#open-links-and-deep-links-universal-links
+    const urlListenerSubscription = Linking.addEventListener(
+      "url",
+      ({ url }) => {
+        // if the url is of this format: iologincie:https://idserver.servizicie.interno.gov.it/idp/login/livello2mobile?value=e1s2
+        // extract the part after iologincie: and dispatch the action to handle the login
+        if (url.startsWith("iologincie:")) {
+          const [, continueUrl] = url.split("iologincie:");
+
+          if (continueUrl) {
+            // https://idserver.servizicie.interno.gov.it/cieiderror?cieid_error_message=Operazione_annullata_dall'utente
+            // We check if the continueUrl is an error
+            if (continueUrl.indexOf("cieiderror") !== -1) {
+              // And we extract the error message and show it in an alert
+              const [, errorMessage] = continueUrl.split(
+                "cieid_error_message="
+              );
```

Comment:
Here you check for `cieiderror` but you split for `cieid_error_message=`.  What about checking also for `cieid_error_message=` before splitting? And what about adding this string to constants? That just make sense to you?

Reply:
While for `profile.html?token=` I'me sure that it's the only possibility, I'm not sure about this kind of URL `https://idserver.servizicie.interno.gov.it/cieiderror?cieid_error_message=Operazione_annullata_dall'utente` propagating an error coming from CieID. In that case if we only check for `cieid_error_message=` in the unlucky scenario where we have a `cieiderror` without `cieid_error_message=` we loose the possibility to show an error screen. What do you think?

Reply:
It makes sense. I will add this check!